server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -627,10 +627,14 @@ public boolean stillLives(long timeLowerBound) {
     return timeIndex.stillLives(timeLowerBound);
   }
 
+  public boolean isDeviceIdInRes(String deviceId) {
+    return timeIndex.checkDeviceIdExist(deviceId);
+  }
+
   /** @return true if the device is contained in the TsFile and it lives beyond TTL */
   public boolean isSatisfied(
       String deviceId, Filter timeFilter, boolean isSeq, long ttl, boolean debug) {
-    if (!timeIndex.checkDeviceIdExist(deviceId)) {
+    if (!isDeviceIdInRes(deviceId)) {
```

Comment:
Hi, here you could directly use timeIndex.checkDeviceIdExist, eliminating a function call